src/ray/raylet/node_manager.cc

```diff
@@ -312,61 +458,133 @@ void NodeManager::SubmitTask(const Task &task, const Lineage &uncommitted_lineag
   }
 }
 
-void NodeManager::AssignTask(const Task &task) {
+void NodeManager::AssignTask(Task &task) {
```

Comment:
Is it an assumption going into `AssignTask` that `task` is currently unqueued?

Reply:
Yeah, I can add that to the documentation.